File: modules/core/src/main/java/kr/entree/enderwand/collection/Maps.java
package kr.entree.enderwand.collection;

import lombok.experimental.UtilityClass;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * Created by JunHyung Lim on 2020-02-04
 */
@UtilityClass
public class Maps {
    public static <K, V> Map<K, V> of() {
        return new LinkedHashMap<>();
    }

    public static <K, V> MapBuilder<K, V> builder() {
        return new MapBuilder<>(of());
    }
}

File: modules/bukkit/src/main/kotlin/kr/entree/enderwand/bukkit/collection/Maps.kt
package kr.entree.enderwand.bukkit.collection

import org.bukkit.command.CommandSender
import org.bukkit.entity.Entity
import java.util.*

/**
 * Created by JunHyung Lim on 2020-01-12
 */
fun <V> MutableMap<UUID, V>.put(entity: Entity, value: V) =
    put(entity.uniqueId, value)

fun <V> MutableMap<String, V>.put(sender: CommandSender, value: V) =
    put(sender.name, value)

fun <V> MutableMap<UUID, V>.remove(entity: Entity) =
    remove(entity.uniqueId)

fun <V> MutableMap<String, V>.remove(sender: CommandSender) =
    remove(sender.name)

operator fun <V> Map<UUID, V>.get(entity: Entity) = get(entity.uniqueId)

fun <V> Map<String, V>.get(sender: CommandSender) = get(sender.name)